### PR TITLE
Project split

### PR DESCRIPTION
**Pull request by adrianstevens**

### CODE REVIEW

File: Source/v2/Meadow.Hcom/Connections/SerialConnection.ListenerProc.cs [hunk 1]
```diff
@@ -238,7 +238,7 @@ private async Task ListenerProc()
                                             throw new Exception("Data received for unknown file");
                                         }
 
-                                        _readFileInfo.FileStream.Write(udp.FileData);
+                                        _readFileInfo.FileStream.Write(udp.FileData, 0, udp.FileData.Length);
```
Comment (CartBlanche) [inline Source/v2/Meadow.Hcom/Connections/SerialConnection.ListenerProc.cs:241]: It's probably an ok change (untested by me), but wondering if this belongs in a separate PR?

**Review by ctacke — APPROVED** [review]

File: Source/v2/Meadow.Cli/Commands/Current/App/AppDeployCommand.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ protected override async ValueTask ExecuteCommand()
             if (!file.Exists)
             {
                 // it's a directory - we need to determine the latest build (they might have a Debug and a Release config)
-                var candidates = PackageManager.GetAvailableBuiltConfigurations(path, "App.dll");
+                var candidates = Package.PackageManager.GetAvailableBuiltConfigurations(path, "App.dll");
```
Comment (stevenkuhn) [inline Source/v2/Meadow.Cli/Commands/Current/App/AppDeployCommand.cs:75]: The `Package.` prefix shouldn't be needed since you are including `Meadow.Package` in the using statement. You should be able to remove this and everywhere else that has the `Package.` prefix.

File: Source/v2/Meadow.Linker/Linker/ILLinker.cs [hunk 3]
```diff
@@ -71,7 +74,9 @@ public ILLinker(ILogger? logger = null)
                     _logger?.Log(LogLevel.Debug, "StandardOutput Contains: " + stdOutReaderResult);
                 }
 
-                await process.WaitForExitAsync();
+                //await process.WaitForExitAsync();
+
+                await Task.Run(() => process.WaitForExit());
```
Comment (stevenkuhn) [inline Source/v2/Meadow.Linker/Linker/ILLinker.cs:79]: What's happening here? I'm guessing `process.WaitForExitAsync()` doesn't exist in `netstandard2.0`? Can the commented out code be removed?

File: Source/v2/Meadow.Package/PackageInfo.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Text.Json.Serialization;
+
+namespace Meadow.Package;
+
+public record PackageInfo
```
Comment (stevenkuhn) [inline Source/v2/Meadow.Package/PackageInfo.cs:5]: You don't _have_ to have this class. You could reference the one in Meadow.Cloud.Client. But I could see the case for decoupling it (even though they are exactly the same right now).

File: Source/v2/Meadow.Package/PackageManager.cs [hunk 5]
```diff
@@ -50,7 +54,7 @@ private bool CleanApplication(string projectFilePath, string configuration = "Re
             if (dataLine.Data != null)
             {
                 Debug.WriteLine(dataLine.Data);
-                if (dataLine.Data.Contains("Build FAILED", StringComparison.InvariantCultureIgnoreCase))
+                if (dataLine.Data.ToLower().Contains("build failed"))
```
Comment (stevenkuhn) [inline Source/v2/Meadow.Package/PackageManager.cs:57]: Why did you change this? The problem with just `ToLower()` is that it is culture dependent. If you are sticking with it, I'd suggest `.ToLower(CultureInfo.InvariantCulture)`. Or go with `dataLine.Data.Contains("build failed", StringComparison.OrdinalIgnoreCase)`.

File: Source/v2/Meadow.Hcom/Connections/SerialConnection.ListenerProc.cs (hunk 1 above)
Comment (adrianstevens) [reply Source/v2/Meadow.Hcom/Connections/SerialConnection.ListenerProc.cs:241]: This is required for the downgrade to .NET Standard 2.0 - otherwise completely agree

File: Source/v2/Meadow.Linker/Linker/ILLinker.cs (hunk 3 above)
Comment (adrianstevens) [reply Source/v2/Meadow.Linker/Linker/ILLinker.cs:79]: exactly - I'll remove the old code

File: Source/v2/Meadow.Package/PackageInfo.cs (hunk 4 above)
Comment (adrianstevens) [reply Source/v2/Meadow.Package/PackageInfo.cs:5]: ah apologies - I meant to delete from Meadow.Cloud.Client

File: Source/v2/Meadow.Package/PackageManager.cs (hunk 5 above)
Comment (adrianstevens) [reply Source/v2/Meadow.Package/PackageManager.cs:57]: the answer is - .NET Standard 2.0 :)

File: Source/v2/Meadow.Cli/Commands/Current/App/AppDeployCommand.cs (hunk 2 above)
Comment (adrianstevens) [reply Source/v2/Meadow.Cli/Commands/Current/App/AppDeployCommand.cs:75]: yeah we have a PackageManager in both HCom and the Package project ...

Comment (adrianstevens) [reply Source/v2/Meadow.Cli/Commands/Current/App/AppDeployCommand.cs:75]: oooh but I can delete the one in HCom

Comment (adrianstevens) [reply Source/v2/Meadow.Cli/Commands/Current/App/AppDeployCommand.cs]: fixed

File: Source/v2/Meadow.Linker/Linker/ILLinker.cs (hunk 3 above)
Comment (adrianstevens) [reply Source/v2/Meadow.Linker/Linker/ILLinker.cs:77]: fixed

File: Source/v2/Meadow.Package/PackageInfo.cs (hunk 4 above)
Comment (adrianstevens) [reply Source/v2/Meadow.Package/PackageInfo.cs:5]: actually you're write - it's in the dependency chain - I deleted the copy in Meadow.Package

File: Source/v2/Meadow.Package/PackageManager.cs (hunk 5 above)
Comment (stevenkuhn) [reply Source/v2/Meadow.Package/PackageManager.cs:58]: Oh my! That makes sense -- I totally thought that would still be available in netstandard2.0.

Thoughts on `.ToLowerInvariant()`?

Comment (adrianstevens) [reply Source/v2/Meadow.Package/PackageManager.cs]: Added CultureInfo.InvariantCulture - thx!

File: Source/v2/Meadow.Dfu/Meadow.Dfu.csproj [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+	<LangVersion>10.0</LangVersion>
+  </PropertyGroup>
```
Comment (stevenkuhn) [inline Source/v2/Meadow.Dfu/Meadow.Dfu.csproj:6]: You need `<Nullable>enable</Nullable>` here and in Meadow.Firmware and in Meadow.Package.

Comment (adrianstevens) [reply Source/v2/Meadow.Dfu/Meadow.Dfu.csproj:6]: great - adding

Comment (adrianstevens) [reply Source/v2/Meadow.Dfu/Meadow.Dfu.csproj:8]: done - added implicit usings while I was in there

**Review by stevenkuhn — APPROVED** [review]
💯
